### ISSUE TITLE
target/file specific cflags heritage

### ISSUE THREAD

**Issue by zbeekman** [szaghi/FoBiS#46]
Hi Stefano,
Would it be possible to make the .cflags.heritage file so that it will keep track of the flags on a per-source and target basis? It may already do this when using a `fobos` file, but when run without a `fobos` each time FoBiS.py is run to build different targets, it overwrites the cflags used for the old target with those used for the new target. That means, that if two targets have different cflags, building either of them will cause FoBiS.py to think that the other one needs to be rebuilt due to stale cflags.

One idea is you could make files like this:
`.cflags.heritage.<target_name>` where `<target_name>` is the name of the thing you’re building.

Another idea is to have one line per target in `.cflags.heritage` with some sort of key at the beginning of the line. Then FoBiS.py can look up the cflags associated with the target it is working on, if they need updating, then rebuild the target and insert the new flags after the key.  You could even just delete the whole line the stale key/flags combo is on if a rebuild is needed due to the flags, and insert the same key with updated flags at the beginning or end of the file.

These lines might look like:
```
libjsonfortran.a : “-O3 -Wextra -Wall -pedantic”
```
where the key is just the target name, followed by a space or some special separator character and then the cflags.

Basically, I think you need a dictionary structure to store cflags at least on a per target basis. It’s possible that you may want to compile different sources for a given target with different flags too, but I’m not sure if FoBiS.py currently supports this, and if it does, I don’t know how to do it.

**Comment by szaghi** [comment]
Hi Izak, i think this should be quite simple. Tomorrow I will try to implement this. Thank you Izaak.

**Comment by szaghi** [comment]
Hi Izaak,

this feature is now into the new version 1.5.8 (on both Github and PyPi). 

With respect your request I have done some modifications:

+ the heritage files are saved per targets, but the name is built as:
    + `build_dir` + target + `.cflags.heritage` if a target is set (presently only one target can be specified for each build, but I think that it can be easily converted to a listed of targets if necessary);
    + `build_dir` + `.cflags.heritage` in case the target is not set.

Note that in your example you have used the `output` for the name of heritage file, but I prefer the `target` value because it is possible to specify a target without the output but not *viceversa*.

I am closing this issue, but you can reopen it if the issue persists.

See you soon.

**Comment by zbeekman** [comment]
Perfecto!
